jl4-core/src/L4/Syntax.hs

```diff
@@ -12,13 +12,18 @@ import Data.Default
 import qualified GHC.Generics as GHC
 import qualified Generics.SOP as SOP
 import Optics
+import qualified Base.Text as Text
+import qualified Data.List.NonEmpty as NE
 
 data Name = MkName Anno RawName
   deriving stock (GHC.Generic, Eq, Show)
   deriving anyclass (SOP.Generic, ToExpr, NFData)
 
 data RawName =
     NormalName Text
+  | QualifiedName Text (NonEmpty Text)
+  -- ^ contains the actual name and a list of qualifiers, e.g.
+  -- foo.bar becomes @'QualifiedName' bar [foo]@
```

Comment:
I see no good reason to flip the order around here, why not keep the qualifiers in front?

Reply:
I don't personally care, so I'll change that.